mesonbuild/dependencies/base.py

```diff
@@ -390,24 +391,21 @@ def find_config(self, versions=None):
         if not isinstance(versions, list) and versions is not None:
             versions = listify(versions)
 
-        if self.env.is_cross_build() and not self.native:
-            cross_file = self.env.cross_info.config['binaries']
-            try:
-                tools = [cross_file[self.tool_name]]
-            except KeyError:
+        for_machine = MachineChoice.BUILD if self.native else MachineChoice.HOST
+        tool = self.env.binaries[for_machine].lookup_entry(self.tool_name)
```

Comment:
This is always a list

Reply:
Nope. It is a list of strings.

Reply:
So, right blow this we check `if tool is None`, that means that `lookup_entry` returns a scalar type, right?

Reply:
Because this calls `stringlistify` for entries, and `shlex.split` or environment variables, `lookup_entry` always returns either `None` or `List[String]`.

`tools` (note the "s") is always a `List[List[String]]`. The outer list is choices, the inner list is command + arguments.

Reply:
ah, okay. that makes sense.